### DISCUSSION TITLE
How to update cardinal and mambaforge3 ?

### DISCUSSION THREAD

**Discussion by Smikehaha** [neams-th-coe/cardinal#762] [General]
Hello, 
How to update cardinal and mambaforge3 ? 
Looking forward to your reply.

**Comment by aprilnovak** [comment]
Hi @Smikehaha - you should be able to just do `conda update --all` to update MOOSE's conda. 

If you want to uninstall and reinstall MOOSE's conda (not typically needed, but in some rare cases it is needed), you can also do:

```
conda deactivate
conda env remove -n moose
conda create -n moose
<go to a new terminal window>
conda activate moose
mamba install moose-libmesh moose-tools
```

To update Cardinal:

```
cd cardinal
git pull upstream devel
./scripts/get-dependencies.sh
```

and then compile as normal. Please let me know if you encounter any issues!

**Reply by Smikehaha** [reply]
'git pull upstream devel' isn't work.
I remove cardinal and re-install it.

**Reply by aprilnovak** [reply]
Ah, I see - I wrote those instructions assuming that the user had set up Cardinal's git remotes according to the instructions [here](https://cardinal.cels.anl.gov/contributing.html). In that command, `upstream` indicates the name of your Cardinal remote. You can figure out what these names are by doing

```
git remote -vv
```

which will print out something like the following

```
april	https://github.com/aprilnovak/cardinal.git (fetch)
april	https://github.com/aprilnovak/cardinal.git (push)
upstream	git@github.com:neams-th-coe/cardinal.git (fetch)
upstream	git@github.com:neams-th-coe/cardinal.git (push)
```

So, you should replace `upstream` with whatever word you see in front of the `neams-th-coe/cardinal.git` remote. If you just cloned the Cardinal repo as-is (without making a fork), the command should be

```
git pull origin devel
```

Sorry for not being more clear!
